tests/testthat/test-fit-mcmc.R

```diff
@@ -192,11 +198,15 @@ test_that("inc_warmup in draws() works", {
   y0 <- fit_mcmc_0$sampler_diagnostics(inc_warmup = FALSE)
   y1 <- fit_mcmc_1$sampler_diagnostics(inc_warmup = FALSE)
   y2 <- fit_mcmc_1$sampler_diagnostics(inc_warmup = TRUE)
+  y3 <- fit_mcmc_3$sampler_diagnostics(inc_warmup = TRUE)
+  y4 <- fit_mcmc_3$sampler_diagnostics(inc_warmup = FALSE)
```

Comment:
Calling sampler_diagnostics() on the same fit would without https://github.com/stan-dev/cmdstanr/pull/293/files#r478645723